nautobot/cloud/models.py

```diff
@@ -97,17 +94,46 @@ def display(self):
         return f"{self.provider}: {self.name}"
 
 
+class CloudResourceTypeMixin(models.Model):
+    """Mixin that designates a model as compatible with CloudResourceType content_types selections."""
+
+    cloud_resource_type = ForeignKeyLimitedByContentTypes(to=CloudResourceType, on_delete=models.PROTECT)
+    extra_config = models.JSONField(null=True, blank=True)
```

Comment:
Moved these fields (and the `clean()` logic below) into the mixin class since they should be common to both `CloudNetwork` and `CloudService`.